### PR TITLE
Add revocation checker for HTTP clients

### PR DESCRIPTION
**Pull request by fredva**
Checks if the incoming server certificate is revoked

### CODE REVIEW

**Comment by fredva** [comment]
🙇

File: src/main/java/no/digipost/security/crl/RevocationChecker.java [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+        } catch (CRLException | IOException e) {
+            throw new RuntimeException(format("Could not load CRL from path '%s'.", crlPath));
+        }
+    }
+
+    @Override
+    public boolean isTrusted(X509Certificate[] chain, String authType) throws CertificateException {
+        for(X509Certificate certificate : chain) {
+            if (crl.isRevoked(certificate)) {
+                throw new CertificateException(format("Certificate with serial number %s is revoked: %s",
```
Comment (runeflobakk) [inline src/main/java/no/digipost/security/crl/RevocationChecker.java]: Bruke [DigipostSecurity.describe(..)](https://github.com/digipost/certificate-validator/blob/master/src/main/java/no/digipost/security/DigipostSecurity.java#L242-L252) for å beskrive sertifikatet?

File: src/main/java/no/digipost/security/crl/RevocationChecker.java [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+                        certificate.getSerialNumber().toString(16),
+                        certificate.getSubjectDN().getName())
+                );
+            }
+        }
+        return false;
```
Comment (runeflobakk) [inline src/main/java/no/digipost/security/crl/RevocationChecker.java]: Når returnerer denne metoden `true`..?

File: src/main/java/no/digipost/security/crl/RevocationChecker.java [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+ *
+ */
+
+public class RevocationChecker implements TrustStrategy {
+
+    private final CRL crl;
+
+    public RevocationChecker(Path crlPath) {
+        CertificateFactory cf = DigipostSecurity.getX509CertificateFactory();
+        try (FileInputStream fis = new FileInputStream(crlPath.toFile())) {
```
Comment (runeflobakk) [inline src/main/java/no/digipost/security/crl/RevocationChecker.java]: Ingen grunn til å konvertere tilbake til gammal `java.io.File`. Bruk heller [Files.newInputStream(crlPath)](https://docs.oracle.com/javase/8/docs/api/java/nio/file/Files.html#newInputStream-java.nio.file.Path-java.nio.file.OpenOption...-)

**Comment by fredva** [comment]
Takk for input @runeflobakk!

**Pull request merged**